org.scala-refactoring.library/src/test/scala/scala/tools/refactoring/tests/util/TestHelper.scala

```diff
@@ -33,27 +36,40 @@ trait TestHelper extends TestRules with Refactoring with CompilerProvider with c
   type GlobalIndexes = analysis.GlobalIndexes
   type ScalaVersion = tests.util.ScalaVersion
 
+  private case class Source(code: String, filename: String) {
+    def toPair = (code, filename)
+  }
+
+  private object Source {
+    def apply(codeWithFilename: (String, String)) = new Source(codeWithFilename._1, codeWithFilename._2)
```

Comment:
Thanks a lot for the hint, I've just tried that out, but I don't think that it is worth changing, as it makes the code only slightly shorter (it requires an additional type annotation) and certainly less readable.